controllers/dbaasprovider_controller.go

```diff
@@ -110,200 +111,3277 @@ func (r *DBaaSProviderReconciler) Reconcile(ctx context.Context, req ctrl.Reques
-					Type:         "int",
-					Required:     true,
-					DefaultValue: strconv.Itoa(defaultAllocatedStorage),
-				},
-				{
-					Name:        iops,
-					DisplayName: "Provisioned IOPS",
-					Type:        "int",
-					Required:    false,
-				},
-				{
-					Name:        maxAllocatedStorage,
-					DisplayName: "Maximum storage threshold",
-					Type:        "int",
-					Required:    false,
+			{
+				Key:         awsSecretAccessKey,
+				DisplayName: "AWS Secret Access Key",
+				Type:        "maskedstring",
+				Required:    true,
+				HelpText:    awsSecretAccessKeyHelpText,
+			},
+			{
+				Key:         awsRegion,
+				DisplayName: "AWS Region",
+				Type:        "string",
+				Required:    true,
+				HelpText:    awsRegionHelpText,
+			},
+			{
+				Key:         ackResourceTags,
+				DisplayName: "ACK Resource Tags",
+				Type:        "string",
+				Required:    false,
+				HelpText:    ackResourceTagsHelpText,
+			},
+			{
+				Key:         ackLogLevel,
+				DisplayName: "ACK Log Level",
+				Type:        "string",
+				Required:    false,
+				HelpText:    ackLogLevelHelpText,
+			},
+		},
+		AllowsFreeTrial:              true,
+		ExternalProvisionURL:         provisionDocURL,
+		ExternalProvisionDescription: provisionDescription,
+		ProvisioningParameters: map[dbaasoperator.ProvisioningParameterType]dbaasoperator.ProvisioningParameter{
+			dbaasoperator.ProvisioningName: {
+				DisplayName: "DB instance identifier",
+				HelpText:    "The name of this instance in the database service.",
+			},
+			dbaasoperator.ProvisioningDatabaseType: {
+				DisplayName: "Engine type",
+				HelpText:    "Select the database engine type for the database instance.",
+				ConditionalData: []dbaasoperator.ConditionalProvisioningParameterData{
+					{
+						Options: []dbaasoperator.Option{
+							{
+								Value:        postgres,
+								DisplayValue: "PostgreSQL",
+							},
+							{
+								Value:        mysql,
+								DisplayValue: "MySQL",
+							},
+							{
+								Value:        mariadb,
+								DisplayValue: "MariaDB",
+							},
+							{
+								Value:        oracleSe2,
+								DisplayValue: "Oracle Standard Edition Two",
+							},
+							{
+								Value:        oracleSe2Cdb,
+								DisplayValue: "Oracle Standard Edition Two with Multitenant",
+							},
+							{
+								Value:        sqlserverEx,
+								DisplayValue: "SQL Server Express Edition",
+							},
+							{
+								Value:        sqlserverWeb,
+								DisplayValue: "SQL Server Web Edition",
+							},
+							{
+								Value:        sqlserverSe,
+								DisplayValue: "SQL Server Standard Edition",
+							},
+							{
+								Value:        sqlserverEe,
+								DisplayValue: "SQL Server Enterprise Edition",
+							},
+						},
+						DefaultValue: postgres,
+					},
 				},
-				{
-					Name:        dbSubnetGroupName,
-					DisplayName: "Subnet group",
-					Type:        "string",
-					Required:    false,
+			},
+			dbaasoperator.ProvisioningMachineType: {
```

Comment:
if we end up doing a ton of maintenance on these lists, we could consider keeping a yaml file to embed or something else to try & ease the pain? Not sure, hopefully we won't need to touch often.

Reply:
Moved to yaml file.

Reply:
There's an alternative method for accomplishing this that we've used in a few places in the operator taking advantage of 'embed' to load the YAML file into a []byte variable that'll unmarshal and not have to worry about the reader. We've used it in a few other places so far. 

Check this out  - https://pkg.go.dev/embed

setup:
https://github.com/RHEcosystemAppEng/dbaas-operator/blob/main/controllers/reconcilers/quickstartinstallation/quickstart_installation_reconciler.go#L21-L31

use:
https://github.com/RHEcosystemAppEng/dbaas-operator/blob/main/controllers/reconcilers/quickstartinstallation/quickstart_installation_reconciler.go#L68-L69

Reply:
Will change it to take this approach later.